### ISSUE TITLE
astroid 2.3.0 is causing RecursionError

### ISSUE THREAD

**Issue by weijiewu8** [PyCQA/pylint#3123]
My code is very long and I'm not sure which part of the code is causing the RecursionError, but the report seems exactly the same as this one: https://github.com/PyCQA/pylint/issues/2734
The code passes pylint until today when pylint is upgraded to 2.4.0 and astroid to 2.3.0. Downgrading pylint to 2.3.0 and astroid to 2.2.5 resolves the issue.

**Comment by asottile** [comment]
seeing this as well, here's a concrete reproduction

I'm seeing this in flake8 https://gitlab.com/pycqa/flake8

The easiest reproduction is:

```bash
git clone git@gitlab.com:pycqa/flake8
cd flake8
git checkout 121ea4f056fabc82d870cf5deaaf37f035be6d10
tox -e pylint
```

It (successfully?) finishes but spews 2300+ lines of output

```console
$ head -20 log
pylint create: /tmp/x/flake8/.tox/pylint
pylint installdeps: pyflakes, pylint
pylint installed: astroid==2.3.0,isort==4.3.21,lazy-object-proxy==1.4.2,mccabe==0.6.1,pyflakes==2.1.1,pylint==2.4.0,six==1.12.0,typed-ast==1.2.0,wrapt==1.11.2
pylint run-test-pre: PYTHONHASHSEED='3707048330'
pylint run-test: commands[0] | pylint src/flake8
Traceback (most recent call last):
  File "/tmp/x/flake8/.tox/pylint/lib/python3.6/site-packages/astroid/__init__.py", line 91, in _inference_tip_cached
    return iter(_cache[func, node])
KeyError: (<function register_builtin_transform.<locals>._transform_wrapper at 0x7f9af9864510>, <Call l.40 at 0x7f9af56b73c8>)

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/tmp/x/flake8/.tox/pylint/lib/python3.6/site-packages/astroid/__init__.py", line 91, in _inference_tip_cached
    return iter(_cache[func, node])
KeyError: (<function register_builtin_transform.<locals>._transform_wrapper at 0x7f9af9864510>, <Call l.40 at 0x7f9af56b73c8>)

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
```

```console
$ tail -20 log
  File "/tmp/x/flake8/.tox/pylint/lib/python3.6/site-packages/astroid/inference.py", line 274, in infer_import_from
    stmts = module.getattr(name, ignore_locals=module is self.root())
  File "/tmp/x/flake8/.tox/pylint/lib/python3.6/site-packages/astroid/scoped_nodes.py", line 534, in getattr
    result = [self.import_module(name, relative_only=True)]
  File "/tmp/x/flake8/.tox/pylint/lib/python3.6/site-packages/astroid/scoped_nodes.py", line 629, in import_module
    return MANAGER.ast_from_module_name(absmodname)
  File "/tmp/x/flake8/.tox/pylint/lib/python3.6/site-packages/astroid/manager.py", line 178, in ast_from_module_name
    return hook(modname)
  File "/tmp/x/flake8/.tox/pylint/lib/python3.6/site-packages/astroid/brain/brain_gi.py", line 134, in _import_gi_module
    raise AstroidBuildingError(modname=modname)
  File "/tmp/x/flake8/.tox/pylint/lib/python3.6/site-packages/astroid/exceptions.py", line 49, in __init__
    super(AstroidBuildingError, self).__init__(message, **kws)
RecursionError: maximum recursion depth exceeded while calling a Python object

--------------------------------------------------------------------
Your code has been rated at 10.00/10 (previous run: 10.00/10, +0.00)

___________________________________ summary ____________________________________
  pylint: commands succeeded
  congratulations :)
```

**Comment by PCManticore** [comment]
Thank you, I can reproduce the issue as well.

**Comment by potiuk** [comment]
Yeah. would love that fixed as well!
